WalletWasabi.Gui/Controls/WalletExplorer/WalletViewModel.cs

```diff
@@ -80,20 +80,33 @@ public WalletViewModel(bool receiveDominant)
 			advancedAction.Items.Add(infoTab);
 			advancedAction.Items.Add(buildTab);
 
-			// Open and select tabs.
+			// Open tabs.
 			sendTab?.DisplayActionTab();
-			if (receiveDominant)
+			receiveTab.DisplayActionTab();
+			coinjoinTab.DisplayActionTab();
+			historyTab.DisplayActionTab();
+
+			// Select tab
+			if (receiveDominant || global.UiConfig.LastActiveTab == receiveTab.GetType().Name)
 			{
-				coinjoinTab.DisplayActionTab();
-				historyTab.DisplayActionTab();
 				receiveTab.DisplayActionTab(); // So receive should be shown to the user.
 			}
-			else
+			else if (global.UiConfig.LastActiveTab == sendTab?.GetType().Name)
+			{
+				sendTab.DisplayActionTab(); // So send should be shown to the user.
+			}
+			else if (global.UiConfig.LastActiveTab == buildTab.GetType().Name)
+			{
+				buildTab.DisplayActionTab(); // So build should be shown to the user.
+			}
+			else if (global.UiConfig.LastActiveTab == historyTab.GetType().Name)
 			{
-				receiveTab.DisplayActionTab();
-				coinjoinTab.DisplayActionTab();
 				historyTab.DisplayActionTab(); // So history should be shown to the user.
 			}
+			else
+			{
+				coinjoinTab.DisplayActionTab(); // So coinjoin should be shown to the user.
+			}
```

Comment:
In these case we prefer switch expressions. It could be something like this:

```c#
var tabToOpen = global.UiConfig.LastActiveTab switch
{
      sendTab.GetType().Name => sendTab,
      receiveTab.GetType().Name => receiveTab,
      .....
      _ => coinjoinTab
};
tabToOpen.DisplayActionTab();
```
```

Reply:
done